data_structures/queue_using_array.cpp

```diff
@@ -1,77 +1,128 @@
-/*
-    Write a program to implement Linear Queue using array.
+/**
+ * @file
+ * @brief Implementation of Linear [Queue using array]
+ * (https://www.geeksforgeeks.org/array-implementation-of-queue-simple/).
+ * @details
+ * The Linear Queue is a data structure used for holding a sequence of
+ * values, which can be added to the end line (enqueue), removed from
+ * head of line (dequeue) and displayed.
+ * ### Algorithm
+ * Values can be added by increasing the `rear` variable by 1 (which points to 
+ * the end of the array), then assigning new value to `rear`'s element of the array.
+ * 
+ * Values can be removed by increasing the `front` variable by 1 (which points to 
+ * the first of the array), so it cannot reached any more.
+ * 
+ * @author [Pooja](https://github.com/pooja-git11)
+ * @author [Farbod Ahmadian](https://github.com/farbodahm)
+ */
+#include <iostream> /// for io operations
+#include <vector>   /// for std::vector
 
-    Functions to implement
-        Enqueue (Insertion)
-        Dequeue (Deletion)
+constexpr uint8_t max_size{10}; ///< Maximum size of the queue
```

Comment:
I think this should be `uint16_t`, to prevent overflows
```suggestion
constexpr uint16_t max_size{10}; ///< Maximum size of the queue
```